tools/tasks.py

```diff
@@ -217,14 +216,17 @@ def __init__(self, description, callable, timeout=None, **kwargs):
 
     def execute(self, fp):
         """Run the task"""
-        print("log_file: {0}. ".format(self.log_file))
+        print(("log_file: {0}. ".format(self.log_file)))
         try:
             result = self.callable()
-            fp.write(result)
+            try:
+                fp.write(result.encode())
+            except (UnicodeEncodeError, AttributeError):
+                fp.write(result)
             return 0
         except Exception as e:
             if self.log_exception:
-                print("Exception executing python task: {0}".format(e))
+                print(("Exception executing python task: {0}".format(e)))
```

Comment:
Almost all of the prints containing only a string don't need double brackets. See example on line 255